src/components/DonationTrackingTable/DonationTrackingTable.jsx

```diff
@@ -29,20 +34,46 @@ const DonationTrackingTable = () => {
   ];
 
   const tabHeaders = ['month', 'quarter', 'year', 'all'];
+  const [currentTab, setCurrentTab] = useState('month'); //change to all once all page is implemented
+  const [currentPageNum, setCurrentPageNum] = useState(1);
+  const [currentDonationsNum, setCurrentDonationsNum] = useState(0);
+  const [pageLimit, setPageLimit] = useState(1);
+  const [searchTerm, setSearchTerm] = useState('');
 
-  const getDonationTrackingTableData = async filter => {
-    const response = await backend.get(`/donation/filter/${filter}`);
-    const data = response.data;
+  const loadInfo = async () => {
+    changePage();
+    const donationNumResponse = await backend.get(
+      `/donation/totalDonations/${currentTab}/?searchTerm=${searchTerm}`,
+    );
+    setCurrentDonationsNum(donationNumResponse.data[0]['count']);
+    setPageLimit(Math.ceil(donationNumResponse.data[0]['count'] / 10));
+    // console.log(donationNumResponse.data.length);
+  };
 
-    for (let i = 0; i < data.length; ++i) {
-      data[i] = { donation_site: 'NULL', ...data[i] };
-    }
+  const changeTab = async tab => {
+    setCurrentTab(tab);
+    setCurrentPageNum(1);
+    setCheckedSet(new Set());
+    setTopCheckBox(false);
+  };
 
-    setDonationTrackingTableData(data);
+  const changePage = async () => {
+    const donationResponse = await backend.get(
+      `/donation/filter/${currentTab}/?pageNum=${currentPageNum}&searchTerm=${searchTerm}`,
+    );
+    setDonationTrackingTableData(donationResponse.data);
   };
+
   useEffect(() => {
-    getDonationTrackingTableData(filter);
-  }, []);
+    const getData = async () => {
+      try {
+        loadInfo(currentTab);
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+    getData();
+  }, [currentTab, currentPageNum, searchTerm]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'loadInfo'. Either include it or remove the dependency array.

Reply:
add load info to dependency array